src/main/java/ecommerce/dao/ProductDaoImpl.java

```diff
@@ -0,0 +1,43 @@
+package ecommerce.dao;
+
+import ecommerce.db.Storage;
+import ecommerce.lib.Dao;
+import ecommerce.model.Product;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+@Dao
+public class ProductDaoImpl implements ProductDao {
+
+    @Override
+    public Product create(Product product) {
+        Storage.addProduct(product);
+        return product;
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        return Storage.products.stream()
+                .filter(prod -> prod.getId() == id)
```

Comment:
According to the checklist we should use wrapper Long for fields. Therefore, it is better to use equals() instead of '=='

Reply:
thank you, done